loggestd/src/log_file.rs

```diff
@@ -23,34 +23,12 @@ fn generate_filename(base_name: &Path, index: usize) -> PathBuf {
     path
 }
 
-fn ensure_directory(directory: &Path) -> Result<(), io::Error> {
-    let result = create_dir(directory);
-
-    match result {
-        Ok(()) => {
-            debug!("Created {}", directory.display());
-        }
-        Err(ref e) if e.kind() == io::ErrorKind::AlreadyExists => (),
-        Err(e) => return Err(e),
-    }
-
-    Ok(())
-}
-
-fn recursive_ensure_directory(directory: &Path) -> Result<(), io::Error> {
-    if let Some(parent) = directory.parent() {
-        recursive_ensure_directory(parent)?;
-    }
-
-    ensure_directory(directory)
-}
-
 impl LogFile {
     pub fn open(base_filename: PathBuf) -> Result<Self, io::Error> {
         let index = 1;
         let filename = generate_filename(&base_filename, index);
 
-        recursive_ensure_directory(filename.parent().unwrap())?;
+        create_dir_all(filename.parent().unwrap())?;
```

Comment:
This change should be in a separate commit, possibly a separate PR.

Reply:
Done